File: arraymedium/majorityelement.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int cnt=0;
        int el;
        for(int i=0;i<nums.size();i++){
            if(cnt==0){
                cnt=1;
                el=nums[i];
            }
            else if(nums[i]==el){
                cnt++;
            }
            else{
                cnt--;
            }
        }
        int ct=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==el){
                ct++;
            }
            if(ct>(nums.size()/2)){
                return el;
            }
        }
        return -1;
    }
};

File: arraymedium/majorityelement.py
class Solution:
    def majorityElement(self, nums):
        count = 0
        candidate = None
        for num in nums:
            if count == 0:
                candidate = num
                count = 1
            elif num == candidate:
                count += 1
            else:
                count -= 1
        count = 0
        for num in nums:
            if num == candidate:
                count += 1
        if count > len(nums) // 2:
            return candidate
        return -1
